app/services/submission/trait_score_parser.rb

```diff
@@ -82,4 +149,22 @@ def csv
   def input
     @input ||= File.open(@upload.file.path)
   end
+
+  def detect_replication(header_columns)
+    header_columns.any?{ |column_name| column_name && column_name.index(/rep\d+$/) }
+  end
+
+  def split_to_trait_and_replicate(column_name)
+    replicate_number = 0
+    return ['',0] if column_name.blank?
+    trait_name = column_name.strip.gsub(/rep\d+$/) do |replicate|
```

Comment:
It took me a while to see what it does :) How about that?
```ruby
matches = column_name.strip.match(/\A(?<trait_name>.+)_rep(?<replicate_number>\d+)\z/)
[matches[:trait_name], matches[:replicate_number]]
```

BTW Doesn't the original code leave trait names suffixed with an underscore?